explainer.md

```diff
@@ -161,119 +160,134 @@ user agent.
-   path, but might have different status, due to either a different query
-   string, or different dynamic context inside the document).
+The reason we have separate APIs is that they have very different concerns which
+are now properly separated:
 
-Number 3 and 4 seem very similar, but architecturally are very different. In
-3, the badge is associated with a URL (and any current or future documents open
-at that URL will show that badge), whereas in 4, the badge is associated with a
-particular document — other tabs open to the same URL will not share the badge.
+* The URL scoping and push message complexity only applies to the scope API.
+* The presence of the document API itself indicates whether the tab will be
+  badged in the UI (obviating the need for a dedicated feature detection method
+  specifically for tab badging).
 
-What this boils down to is two different modes:
+Other reasons to separate:
 
-* Badging a particular URL scope, applied to the set of URLs that start with the
-  same path. This covers 1, 2 and 3 above.
-* Badging a particular document, applied to only the browser tab containing the
-  document.
+* Sites that only want to badge their app icon or only want to badge their tab
+  can just concern themselves with the relevant API.
+* The document API can potentially support badges that aren't supported by host
+  operating systems, such as arbitrary Unicode characters.
 
-The API for requesting to badge the current document, as opposed to a URL set,
-is still under consideration. It may simply be a separate option member, e.g.:
+User agents may provide either or both of the two APIs, so sites should
+feature-detect them individually. In particular, this allows you to fall back to
+showing your own badge in the page favicon or title if the
+`Badge.setForDocument` API is not available (regardless of the availability of
+the scope API). This is a complete example for a site that wants to set the
+badge on both the current document and for apps/handles across the origin:
 
 ```js
-Badge.set(getUnreadCount(), {scopeDocument: true});
+// Should be called whenever the unread count changes (new mail arrives, or mail
+// is marked read/unread).
+function unreadCountChanged(newUnreadCount) {
+  // Set the "handle" badge, for app icons and links. This has a global and
+  // semi-permanent effect, outliving the current document.
+  if (Badge && Badge.setForScope) {
+    Badge.setForScope(newUnreadCount);
+  }
+
+  // Set the "document" badge, for the current tab / window icon.
+  if (Badge && Badge.setForDocument) {
+    Badge.setForDocument(newUnreadCount);
+  } else {
+    // Fall back to setting favicon (or page title).
+    showBadgeOnFavicon(newUnreadCount);
+  }
+}
 ```
 
-When displaying the badge for an app, the user agent should use the badge
-matching the app's [scope URL](https://www.w3.org/TR/appmanifest/#scope-member).
+More advanced examples are given in a [separate document](docs/examples.md).
```

Comment:
You probably need to update examples.md too

Reply:
I'll do that later, depending on whether we commit to this.